File: files/2024-06-16/CSEC 16.06.py
import pandas as pd
import numpy as np

path = "files/Excel Challenge 16th June.xlsx"
input = pd.read_excel(path, skiprows=1, usecols="B")
test = pd.read_excel(path, skiprows=1, usecols="D:E", nrows = 3)

result = pd.DataFrame(input.values.reshape(-1, 2), columns=['A', 'B'])
result['month'] = result['A'].dt.month
result = result.groupby('month').tail(1)
result = result[['month', 'B']].reset_index(drop=True)
result = result.astype('int64')

result.columns = test.columns

print(result.equals(test)) # True

File: files/2024-06-16/CSEC 16.06.R
library(tidyverse)
library(readxl)

path = "files/Excel Challenge 16th June.xlsx"
input = read_excel(path, range = "B2:B22")
test  = read_excel(path, range = "D2:E5")

result = input %>%
  as.list() %>%
  unlist() %>%
  matrix(ncol = 2, byrow = TRUE) %>%
  as_tibble() %>%
  setNames(c("Date", "Value")) %>%
  mutate(Date = as.Date(Date, origin = "1899-12-30"),
         Month = month(Date)) %>%
  summarise(`Closing Order` = last(Value), .by = Month) 

identical(result, test)
# [1] TRUE